### PR TITLE
Vagrant, Sentry, Vault, cleanup

### PR DESCRIPTION
**Pull request by spladug**
Pretty much what it says on the tin. The ugliness in the handler makes me pretty nauseous so if anyone's got a suggestion for something better please let me know!

### CODE REVIEW

File: reddit_service_websockets/socketserver.py [hunk 1]
```diff
@@ -46,6 +45,16 @@ def log_request(self):
         # geventwebsocket logs every request at INFO level. that's annoying.
         pass
 
+    @property
+    def socket_server(self):
+        # this is a pretty nasty smell. we're violating the wsgi model by
+        # expecting the application callable to be of a specific type so that
+        # we can get access to its state for various things (metrics, secrets).
+        # this property at least gets us a little encapsulation.
+        socket_server = self.application.application
```
Comment (ckwang8128) [inline reddit_service_websockets/socketserver.py]: I'm confused by the second level of accessing here (`application.application`), how does this work?

**Comment by ckwang8128** [comment]
I can't think of a better way around the handler, but I do have a question about how it's working. Otherwise looks good! :nail_care:

File: reddit_service_websockets/socketserver.py (hunk 1 above)
Comment (spladug) [inline reddit_service_websockets/socketserver.py]: `Sentry` is a WSGI middleware that wraps our application. This grabs the inner application from the outer.

https://github.com/getsentry/raven-python/blob/master/raven/middleware.py#L86-L87

Comment (ckwang8128) [inline reddit_service_websockets/socketserver.py]: Ah got it, thanks!

**Review by ckwang8128 — APPROVED** [review]

**Comment by dellis23** [comment]
🐟 Exciting!

**Pull request merged**